### DISCUSSION TITLE
Ability to create local users easily

### DISCUSSION THREAD

**Discussion by ngerritsen** [supabase/supabase#11828] [Questions]
# Feature request

## Is your feature request related to a problem? Please describe.

When developing locally it can be a bit of a hassle to get a user to work. Even if you have a whole sign up flow, you still have to fiddle around with the Inbucket and it's not very clear how it works. Also when developing locally I like to have a script that initializes my whole environment (kinda like the `seed.sql` and `migrations.sql` are doing for the database), it would be very convenient if you could also easily populate the database with some test users.

## Describe the solution you'd like

1. Better documentation on the currrent situation. I had to do quite some digging to find out how to create a user locally without a signup flow and how I could get my email from Inbucket. It might be nice to have some docs on that (especially the email part with the _@inbucket.org_ adresses).
2. Have some convenient way to create test users locally (maybe with the Supabase CLI).

## Describe alternatives you've considered

I called the `auth.api.createUser` endpoint using the server only key to create a user, then went to Inbucket and confirmed the user (which only worked with an _@inbucket.org_ email). My app prototype does not a have a full sign up flow yet which I guess is going to be the case for more apps.

## Additional context

These are just suggestions and things I noticed. I'm curious on other views on this. I think at least "something" should be done to improve the local development with users.

**Comment by gwvt** [comment]
@ngerritsen I agree it would be really nice to be able to do this with the CLI. As a workaround you can create test users manually with your UI or the Supabase client, then run:
```
pg_dump postgresql://postgres:postgres@localhost:54322/postgres --inserts -f dump.sql
```
From the dump.sql file copy and paste the INSERT INTO auth.users ... and INSERT INTO auth.identities ... queries into supabase/seed.sql.

**Reply by philipbel** [reply]
Thanks for sharing the tip. Rather than dealing with roles (`postgres` user can't use `pg_dump`), I used

```
supabase db dump --db-url postgresql://postgres:postgres@localhost:54322/postgres -s auth --data-only
```

**Comment by hf** [comment]
When you say Inbucket, are you referring to: https://inbucket.org?

Note that you do need a real SMTP server to send emails nowadays. Make sure you setup the docker SMTP settings to point to your Inbucket deployment correctly. Example configs:

```
GOTRUE_SMTP_HOST=smtp.mandrillapp.com
GOTRUE_SMTP_PORT=587
GOTRUE_SMTP_USER=smtp-delivery@example.com
GOTRUE_SMTP_PASS=correcthorsebatterystaple
GOTRUE_SMTP_ADMIN_EMAIL=support@example.com
```

I'll convert this issue to a discussion as there's not much that can be improved when dealing with SMTP.

If you have experience with Inbucket, given it appears to be open-source, do you think it may be worthwhile to include it in our docker setup so it just works?

**Comment by evanfuture** [comment]
Just to add that Supabase local already uses Inbucket to capture any outgoing emails from the local setup.  When you run `supabase start` it's one of the url's you get, something like `Inbucket URL: http://localhost:54324`

**Reply by jahzielv** [reply]
idk if this is present anywhere in the docs, but thanks so much for pointing this out!! was beating my head against the wall trying to figure out where emails were going lol

**Comment by jammygrams** [comment]
Writing up manual user creation for dummies, in case you're like me.  Once you have Supabase running locally,
1. View the local instance of the UI at `http://localhost:54323/`
2. Go to "Authentication" and hit "+ Invite" to create a user. Use @inbucket.org domain for the email e.g. "jay@inbucket.org "
3. Go to the Inbucket page at `http://localhost:54324/` and search for the username you used (e.g. "jay") to get the inbox of the user
4. Hit "accept the invite" and you'll get the user token in the url of the subsequent page

**Reply by ledjainam** [reply]
What if we are using google auth?
